src/Infrastructure/EntityObserver/ProcessLowestPriceOnCheckingPeriodChangeObserver.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Infrastructure\EntityObserver;
+
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\PriceHistoryPlugin\Application\Processor\ProductLowestPriceBeforeDiscountProcessorInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+
+final class ProcessLowestPriceOnCheckingPeriodChangeObserver implements EntityObserverInterface
+{
+    public function __construct(
+        private ProductLowestPriceBeforeDiscountProcessorInterface $productLowestPriceBeforeDiscountProcessor,
+        private RepositoryInterface $channelPricingRepository,
+    ) {
+    }
+
+    public function onChange(object $entity): void
+    {
+        if (!$entity instanceof ChannelInterface) {
+            return;
```

Comment:
Should throw an exception since the proposed architecture relies on the usage of `getSupportedEntity`.

Reply:
Also `getSupportedEntity` should be used for this check

Reply:
I did it only because of the static analysis as simple assertion didn't fix the issue however you are right. I'll try to deal with it differently.